apps/meteor/app/livechat/server/lib/QueueManager.ts

```diff
@@ -246,7 +246,14 @@ export class QueueManager {
 			void notifyOnSettingChanged(livechatSetting);
 		}
 
-		const newRoom = (await this.queueInquiry(inquiry, room, defaultAgent)) ?? (await LivechatRooms.findOneById(rid));
+		let newRoom;
+		if (room.v.contactId && (await shouldTriggerVerificationApp(room.v._id, room.v.contactId, room.source))) {
+			newRoom = await LivechatRooms.findOneById(rid);
+			await LivechatContacts.updateContactChannel({ visitorId: room.v._id, source: room.source }, { verified: false });
```

Comment:
Should this be on a model method? I mean, the `verified: false` part, should it be _inside_ a model method? Something like `setContactUnverified` (to follow the same pattern on other models) ?

wdyt?

Reply:
I don't know if I agree with this one. We're just updating a channel, having an exclusive method to update a single property seems a bit too much. Also, we would need a second one to `setContactVerified`, and then other for the `block` property, other for the `unknown`...

Reply:
Which is why we have a bunch of model methods 😬 

But generally, we can do a `toggleContactVerification` method that can handle both verified states, wdyt?

Reply:
Brought this to the SCI team (internally) and to @MarcosSpessatto and we agreed that even though splitting this method in multiple ones has some beneficial aspects (such as clarity and more "precisely" defined single responsibility for these updates), we'd end up losing on the performance side due to more DB trips needed (which is something we're trying to avoid), which could go even worse in the future since we don't know exactly how this feature might scale.
So we decided to keep this method for all updates as of now

Please reopen this suggestion if I didn't get your point correctly or in case you have another suggestion tthe downsides I mentioned, @KevLehman

Reply:
wait, how come moving the func to the model will cause more db trips? Maybe i'm missing something. 

My idea was to just hide the logic in the model, something like:
```
Contacts.setContactVerified({})
```

So the `verified: true` part is inside the model, as we do for a bunch of other functions. Not sure how that would affect perf 🤔

Reply:
> wait, how come moving the func to the model will cause more db trips? Maybe i'm missing something.

In this specific case you mentioned for blocked, there's no difference/problem at all. We could do that, but I wanted to bring the full picture so that we don't make thing inconsistent

In of the cases where `updateContactChannel` is called, there are more fields being set together and they're not exactly related. so we had three options: (1) create a weird method that sets a bunch of unrelated (possibly undefined) fields together, (2) split it in some more functions and increase db trips -> affect performance OR (3) create more specific methods for other cases (eg the verified one you mentioned), but keep this one that sets many fields as it is now -- but that feels somehow inconsistent, doesn't it?